pkg/resource/manager.go

```diff
@@ -103,14 +113,31 @@ func NewResourceManager(ctx context.Context, resourceNames []string, wrapper api
 }
 
 func (m *Manager) GetResourceProviders() map[string]provider.ResourceProvider {
+	m.lock.RLock()
```

Comment:
I guess you don't need lock here, right? This function creates the map and the map will be maintained and shared with goroutines. Seems no reason to lock the map creation. Can you double check if this is called once when creating manager? Also this is a `Set` function instead of `Get`. Can you rename it? Thanks.

Reply:
I think map creation happens here: https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/0c6381a1ef46b1315dbe3b88d453115a1623e88f/pkg/resource/manager.go#L107-L110

Here is making a copy of the original map and returning the copy, so I guess it still counts as a `Get`?

Reply:
hmm, I see. May I ask the reason to make a copy there? I don't see an update/modification here? Since we have small PRs, I am not able to check what the function is for.

Reply:
https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/0e53849b687af632c490bf3c1b3cf3a69d7abcaa/pkg/resource/manager.go#L125-L131

I'm not sure why the original author made a copy of the map here, but I'm guessing the original map was not returned in case it could be modified by the caller?

Reply:
Realized this is not a copy of `resource` which contains both provider and handler, it's a new map from resource name to resource provider.